rotkehlchen/db/filtering.py

```diff
@@ -568,6 +569,30 @@ def prepare(self) -> tuple[list[str], list[T]]:
         )
 
 
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class DBMultiStringAsFloatRangeFilter(DBFilter):
+    """Filters a column having a string value that represents a decimal, out of a selection of
+    decimal values. It also allows a tolerance for the comparison."""
+
+    column: str
+    values: list[FValWithTolerance]
+
+    def prepare(self) -> tuple[list[str], list[str]]:
+        conditions = []
+        parameters = []
+
+        for val in self.values:
+            value = val[0]
+            tolerance = val[1] * FVal(0.5)
+            conditions.append(f'CAST({self.column} AS REAL) BETWEEN ? AND ?')
+            parameters.extend([str(value - tolerance), str(value + tolerance)])
+
+        return (
+            [' OR '.join(conditions)],
```

Comment:
Why OR? Shouldn't it be AND? Or at least make this configurable? It already is an argument and you are giving it. The `and_op=True` one.

Reply:
It needs to be OR in this case because we query for a list of amounts. here is an example: 
![image](https://github.com/rotki/rotki/assets/12560596/9d94d40b-0811-4d59-8dac-25a29c2f6c0d)
That being said, I updated the filter to make the operation (AND or OR) configurable using the and_op parameter.

Reply:
In that case there is a problem. This query should match each individual claim event. it does not.
Could also potentially match mistaken events if another event has same asset or any other combination of things here.


 I don't think our filters can work here. Cause you are trying to apply a multifilter. I really think raw sqlite is better here as I had written on Friday.